src/sessions/VirtualPageLoadTimer.ts

```diff
@@ -168,7 +168,7 @@ export class VirtualPageLoadTimer extends MonkeyPatched {
             .catch((error) => {
                 throw error;
             })
-            .finally(self.decrementFetchCounter());
+            .finally(self.decrementFetchCounter.bind(self));
```

Comment:
```suggestion
            .finally(() => self.decrementFetchCounter());
```

Reply:
Same idea.

Reply:
I'm trying to propose something [more idiomatic](https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions). However I think my prior suggestion is incorrect. We can make `decrementFetchCounter` an arrow function, and then we do not need to bind. I don't think we need `self` here either, since `fetch` is an arrow function.
```suggestion
            .finally(this.decrementFetchCounter);
```

```typescript
    private decrementFetchCounter = () => {
        if (!this.isPageLoaded) {
            this.latestEndTime = Date.now();
        }
        this.fetchCounter -= 1;
    };
```

Reply:
Sounds good to me :-)

Reply:
Complete